File: 01a [Individual] Data parsing servers - Part I/data_parsing.py
import json
from pathlib import Path

BASE_FOLDER = Path(__file__).parent.parent / "02_Text-based_Data_Formats"
file_types = [".json", ".csv", ".xml", ".yaml", ".txt"]

def parse_file(file_path):
    data = file_path.read_text(encoding="utf-8")
    return json.dumps(json.loads(data), indent=4) if file_path.suffix == ".json" else data

def print_data_structure(data):
    print(data)

def main():
    for ext in file_types:
        file_path = BASE_FOLDER / f"me{ext}"
        print(f"\n--- {file_path.name} ---")
        print_data_structure(parse_file(file_path))

main()

File: 01a [Individual] Data parsing servers - Part I/data_parsing.js
const fs = require("fs");
const path = require("path");

const BASE_FOLDER = path.join(__dirname, "..", "02_Text-based_Data_Formats");
const fileTypes = [".json", ".csv", ".xml", ".yaml", ".txt"];

function parseFile(filePath, ext) {
    const data = fs.readFileSync(filePath, "utf-8");

    if (ext === ".json") {
        return JSON.stringify(JSON.parse(data), null, 4);
    }
    return data;
}

function printDataStructure(data) {
    console.log(data);
}

function main() {
    fileTypes.forEach(ext => {
        const filePath = path.join(BASE_FOLDER, `me${ext}`);

        console.log(`\n--- ${path.basename(filePath)} ---`);
        const parsedData = parseFile(filePath, ext);
        printDataStructure(parsedData);
    });
}

main();